module_properties/utils/properties.py

```diff
@@ -0,0 +1,253 @@
+        return data
+    else:
+        raise InvalidRequestError(
+            "Invalid Request - Request on InChi Key "
+            + inchikey
+            + " failed with status code "
+            + response.status_code
+        )
+
+
+def insert_properties_from_smiles(smiles, properties=["All"]):
+    try:
+        m = Chem.MolFromSmiles(smiles)
+    except:
+        raise InvalidSMILESError(
+            "Invalid SMILES - The provided SMILES "
+            + smiles
+            + " cannot be converted into a valid molecule"
+        )
+
+    inchikey = Chem.MolToInchiKey(m)
+    prop = Properties(id=inchikey)
+    prop.inchikey = inchikey
+    try:
+        prop.validate()
+    except:
+        raise InChiKeyNotFoundError(
+            "InChI Key Not Found - The provided InChI Key "
+            + inchikey
+            + " was not found in the compounds collection of the database"
+        )
+
+    url = __format_request_url(inchikey, properties)
+    data = __make_url_request(inchikey, url)
+    data = data["PropertyTable"]["Properties"][0]
+
+    pc = Pubchem()
+    pc.MolecularFormula = data["MolecularFormula"]
+    pc.MolecularWeight = data["MolecularWeight"]
+    pc.XLogP = data["XLogP"]
+    pc.ExactMass = data["ExactMass"]
+    pc.MonoisotopicMass = data["MonoisotopicMass"]
+    pc.TPSA = data["TPSA"]
+    pc.Complexity = data["Complexity"]
+    pc.Charge = data["Charge"]
+    pc.HBondDonorCount = data["HBondDonorCount"]
+    pc.HBondAcceptorCount = data["HBondAcceptorCount"]
+    pc.RotatableBondCount = data["RotatableBondCount"]
+    pc.HeavyAtomCount = data["HeavyAtomCount"]
+    pc.IsotopeAtomCount = data["IsotopeAtomCount"]
+    pc.AtomStereoCount = data["AtomStereoCount"]
+    pc.DefinedAtomStereoCount = data["DefinedAtomStereoCount"]
+    pc.UndefinedAtomStereoCount = data["UndefinedAtomStereoCount"]
+    pc.BondStereoCount = data["BondStereoCount"]
+    pc.DefinedBondStereoCount = data["DefinedBondStereoCount"]
+    pc.UndefinedBondStereoCount = data["UndefinedBondStereoCount"]
+    pc.CovalentUnitCount = data["CovalentUnitCount"]
+    pc.Volume3D = data["Volume3D"]
+    pc.XStericQuadrupole3D = data["XStericQuadrupole3D"]
+    pc.YStericQuadrupole3D = data["YStericQuadrupole3D"]
+    pc.ZStericQuadrupole3D = data["ZStericQuadrupole3D"]
+    pc.FeatureCount3D = data["FeatureCount3D"]
+    pc.FeatureAcceptorCount3D = data["FeatureAcceptorCount3D"]
+    pc.FeatureDonorCount3D = data["FeatureDonorCount3D"]
+    pc.FeatureAnionCount3D = data["FeatureAnionCount3D"]
+    pc.FeatureCationCount3D = data["FeatureCationCount3D"]
+    pc.FeatureRingCount3D = data["FeatureRingCount3D"]
+    pc.FeatureHydrophobeCount3D = data["FeatureHydrophobeCount3D"]
+    pc.ConformerModelRMSD3D = data["ConformerModelRMSD3D"]
+    pc.EffectiveRotorCount3D = data["EffectiveRotorCount3D"]
+    pc.ConformerCount3D = data["ConformerCount3D"]
+    pc.Fingerprint2D = data["Fingerprint2D"]
+
+    rd = RDKit()
+    rd.cipher_mid = "002"
```

Comment:
is this supposed to be constant?

Reply:
This will be a constant for what ever MID is given to RDKit once we decide on an identification structure for the models collection of the database. I can declare it a constant above and then change it if needed